### PR TITLE
test overcloud nodes hostname

### PR DESCRIPTION
**Pull request by rrasouli**
Checking the smbios details of hostname in devices section
{{pytest: cfme/tests/infrastructure/test_host_hostname.py -v --use-provider tripleo}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 37409e8e3df0cc901b80507b06d3e0b4d33b200f:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/tests/infrastructure/test_host_hostname.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_host_hostname.py]: Unnecessary blank line at start, you can put:
```python
# -*- coding: utf-8 -*-
```
at the beginning though.

File: cfme/tests/infrastructure/test_host_hostname.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = \
+        testgen.infra_providers(metafunc, required_fields=["ssh_credentials"])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist,
+                        scope="module")
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_host_hostname.py]: you an mark the tests globally in this module using

```python
pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
```

File: cfme/tests/infrastructure/test_host_hostname.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+from cfme.fixtures import pytest_selenium as sel
+from cfme.configure.tasks import is_host_analysis_finished
+from utils.wait import wait_for
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = \
+        testgen.infra_providers(metafunc, required_fields=["ssh_credentials"])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist,
+                        scope="module")
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_host_hostname.py]: Replace this whole function by:

```python
pytest_generate_tests = testgen.generate(testgen.provider_by_type, ['openstack-infra'], scope='module')
```

File: cfme/tests/infrastructure/test_host_hostname.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_host_hostname(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_host_hostname.py]: You can replace these two lines by:

```python
provider.summary.relationship.nodes.click()
```

Which will load the summary page and will allow you to click through.

File: cfme/tests/infrastructure/test_host_hostname.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    for quad in my_quads:
+        host = Host(name=quad.name)
+        host.run_smartstate_analysis()
+        wait_for(lambda: is_host_analysis_finished(host.name), delay=15,
+                 timeout="10m", fail_func=lambda: toolbar.select('Reload'))
+        result = host.get_detail("Properties", "Hostname")
+        assert result != ''
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_host_hostname.py]: You don't check whether the hostname was collected correctly? Just that it was collected?

Comment (rrasouli) [inline cfme/tests/infrastructure/test_host_hostname.py]: Correct just checking at this phase if the hostname collected

**Comment by dajoRH** [comment]
Lint report for commit 99a1c3609a6e1dda23c65c63e0d5b1197e206594:

`cfme/tests/infrastructure/test_host_hostname.py`:
- :bangbang: Line 11:1: **E302** *expected 2 blank lines, found 0*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fc542a8012b3486cdf403c87a8d0fb7177bc4665:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/tests/infrastructure/test_host_hostname.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+from cfme.configure.tasks import is_host_analysis_finished
+from utils.wait import wait_for
+import pytest
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack-infra'], scope='module')
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+def test_host_hostname(provider):
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_host_hostname.py]: Can you also add a docstring? Like https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/cli/test_rake_evm_automate-53.py#L38

The contents of docstrings get uploaded to Polarion once the test runs. And we want it to be on each test.

**Comment by dajoRH** [comment]
Lint report for commit 092fe51633362a9d42bfe976340f566116b69b33:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a420152f18d1fc2ce03e810f8862a8791d7c4d67:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by rrasouli** [comment]
@dajoRH please merge

**Comment by dajoRH** [comment]
Response to comment #245641673:

It seems like all the basic requirements were met, I have therefore merged your PR.

**Pull request merged**